hs_core/models.py

```diff
@@ -1668,6 +1666,33 @@ class MyResourceContentType(pages.Page, hs_core.AbstractResource):
     # this field WILL NOT get recorded in bag and SHOULD NEVER be used for storing metadata
     extra_data = HStoreField(default={})
 
+    @property
+    def view_count(self):
+        return int(self.extra_data.get('view_count', '0'))
+
+    @property
+    def download_count(self):
+        return int(self.extra_data.get('download_count', '0'))
+
+    def update_view_count(self, request):
```

Comment:
I don't get this method. I see three routes here:
1. the request has resource_ids
  a. the resource id is not in the resource_ids - updates the view count
  b. the resource id is in the resource_ids - the view count is not update (why not?)
2. the request does not have resource_ids - the view count is updated.

I'm not following why this method does and does not update the view_count.  And I believe the request object has no business being in here.